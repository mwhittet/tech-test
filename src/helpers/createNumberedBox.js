import { handleClick } from "../numberSelection";

export const createNumberedBox = number => {
  let box = document.createElement("div");
  let text = document.createTextNode(number);

  box.setAttribute("class", "box");
  box.onclick = () => handleClick(number);
  box.appendChild(text);

  return box;
};
