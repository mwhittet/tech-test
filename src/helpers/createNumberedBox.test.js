import { createNumberedBox } from "./createNumberedBox";

jest.mock("../numberSelection", () => ({
  handleClick: jest.fn()
}));

describe("helpers > createNumberedBox", () => {
  it("returns a div with a numeric value", () => {
    const number = 99;
    const result = createNumberedBox(number);

    expect(result).toMatchSnapshot();
    expect(result.innerHTML).toBe(`${number}`);
  });
});
