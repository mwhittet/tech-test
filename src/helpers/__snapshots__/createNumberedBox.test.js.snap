// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers > createNumberedBox returns a div with a numeric value 1`] = `
<div
  class="box"
>
  99
</div>
`;
